### PR TITLE
Marking some core types @BetaApi and @InternalApi

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**
Types that will go GA:

- spi/ServiceRpcFactory
- BaseService
- BaseServiceException
- BaseWriteChannel
- BatchResult
- ExceptionHandler
- FieldSelector
- GcpLaunchStage
- HttpTransportOptions
- Identity
- MonitoredResource
- MonitoredResourceDescriptor
- NoCredentials
- Policy
- ReadChannel
- Restorable
- RestorableState
- RetryHelper
- Role
- Service
- ServiceDefaults
- ServiceFactory
- ServiceOptions
- ServiceRpc
- Timestamp
- TransportOptions
- WriteChannel

### CODE REVIEW

File: google-cloud-core/src/main/java/com/google/cloud/GrpcTransportOptions.java [hunk 1]
```diff
@@ -41,6 +41,7 @@
  * Class representing service options for those services that use gRPC as the transport
  * layer.
  */
+@BetaApi
```
Comment (michaelbausor) [inline google-cloud-core/src/main/java/com/google/cloud/GrpcTransportOptions.java]: This isn't going GA with APIs that use it?

Comment (garrettjonesgoogle) [inline google-cloud-core/src/main/java/com/google/cloud/GrpcTransportOptions.java]: Oh, this should actually go GA with Logging. I'll remove the annotation.

**Comment by garrettjonesgoogle** [comment]
Updated, PTAL

**Comment by michaelbausor** [comment]
LGTM once test issues are resolved

**Pull request merged**